src/Microsoft.Framework.ConfigurationModel/Configuration.cs

```diff
@@ -136,14 +136,29 @@ public IConfiguration GetSubKey(string key)
                 new ConfigurationFocus(this, prefix + segment + Constants.KeyDelimiter));
         }
 
+        /// <summary>
+        /// Adds a new configuration source
+        /// </summary>
+        /// <param name="configurationSource">The configuration source to add</param>
+        /// <returns>The same configuration source</returns>
         public IConfigurationSourceRoot Add(IConfigurationSource configurationSource)
         {
-            configurationSource.Load();
-            return AddLoadedSource(configurationSource);
+            return Add(configurationSource, load: true);
         }
 
-        internal IConfigurationSourceRoot AddLoadedSource(IConfigurationSource configurationSource)
+        /// <summary>
+        /// Adds a new configuration source
+        /// </summary>
+        /// <param name="configurationSource">The configuration source to add</param>
+        /// <param name="load">If true, the configuration source will be loaded upon add</param>
```

Comment:
Could possibly be more explicit here too: "the configuration source's {magical_doc_ref:IConfigurationSource::Load} method will be called"